test/test_greedy_star.jl

```diff
@@ -0,0 +1,83 @@
+using SparseDiffTools
+using LightGraphs
+using Random
+
+Random.seed!(123)
+
+#= Test data =#
+test_graphs = Array{SimpleGraph, 1}(undef, 0)
+
+for _ in 1:5
+    nv = rand(5:20)
+    ne = rand(1:100)
+    graph = SimpleGraph(nv)
+    for e in 1:ne
+        v1 = rand(1:nv)
+        v2 = rand(1:nv)
+        while v1 == v2
+            v2 = rand(1:nv)
+        end
+        add_edge!(graph, v1, v2)
+    end
+    push!(test_graphs, copy(graph))
+end
+
+#=
+ Coloring needs to satisfy two conditions:
+
+1. every pair of adjacent vertices receives distinct  colors
+(a distance-1 coloring)
+
+2. For any vertex v, any color that leads to a two-colored path
+involving v and three other vertices  is  impermissible  for  v.
+In other words, every path on four vertices uses at least three
+colors.
+=#
+
+for i in 1:5
+    g = test_graphs[i]
+
+    out_colors1 = SparseDiffTools.color_graph(g,SparseDiffTools.GreedyStar1Color())
+    out_colors2 = SparseDiffTools.color_graph(g,SparseDiffTools.GreedyStar2Color())
+
+    #test condition 1
+    for v = vertices(g)
+        color = out_colors1[v]
+        for j in inneighbors(g, v)
+            @test out_colors1[j] != color
+        end
+    end
+
+    #test condition 2
+    for j = vertices(g)
+        walk = saw(g, j, 4)
```

Comment:
Given the weird and ambiguous name of the function, it might be better to make it explicit
```suggestion
        walk = LightGraphs.saw(g, j, 4)
```